client/ayon_blender/api/pipeline.py

```diff
@@ -379,9 +381,42 @@ def on_open():
                 "Base file unit scale changed to match the project settings.")
 
 
+def on_before_save(event):
+    """Handle the event before saving a Blender file.
+
+    When saving to a new scene name, e.g. on incrementing the workfile then
+    update the render paths to match the new scene name by replacing the old
+    scene name with the new one in all render paths.
+    """
+    blend_path_before: str = bpy.data.filepath
+    blend_path_after: str = event.get("filename")
+
+    if not blend_path_before:
+        # Saving from a new unsaved file, no need to check for changes.
+        return
+
+    blend_name_before: str = os.path.splitext(
+        os.path.basename(blend_path_before))[0]
+    blend_name_after: str = os.path.splitext(
+        os.path.basename(blend_path_after))[0]
+    if blend_name_before != blend_name_after:
+        print(f"Detected scene name change from {blend_name_before} to "
+              f"{blend_name_after}")
+        # TODO: We may want to allow disabling this feature, especially after
+        #  Blender 4.5+ supporting the `{blend_name}` in templates in render
+        #  paths to avoid needing to hardcode the blender scene filename into
+        #  those paths.
+        # Update any render paths if they previously contained the scene name
+        # Warning: if the scene name is `a` before and now `helloworld` then
+        #  this may easily get out of hand by turning `asset` into
+        #  `helloworldsset`, but since filenames tend to be longer and
+        #  contain version numbers, this is not expected to happen often.
+        search_replace_render_paths(blend_name_before, blend_name_after)
```

Comment:
@moonyuet @antirotor @philippe-ynput this makes it very convenient - because the user doesn't need to care about manipulating the render paths each time, or having it invalidated on publish to just keep clicking 'repair' each time you publish. However, as the `warning` suggests in the comment there may be edge cases where this may _not_ do what one would want. Thoughts?

Reply:
It's a good idea but I dont think we won't need to include in every blender scene as some of them are only for building content/animation only. Would it be better to be just introduced when there is file ouput node lasted in the scene?

Reply:
> In that case the paths likely wouldn't be set anyway - so there it wouldn't actually be replacing anything... it'd just end up doing nothing. So essentially the behavior is the same as if I would skip it by doing custom checks?

Ahh...My bad, I just read the related function again that you already have skipped for other shader nodes for the searching part. I am perfectly find with this callback.